### ISSUE TITLE
A question about DiceLoss

### ISSUE THREAD

**Issue by zhongqiu1245** [open-mmlab/mmsegmentation#1266]
Hi, dear authors!
Thank you for your amazing job!
I have a question about your DiceLoss.
In dice_loss.py, line 106:
https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/models/losses/dice_loss.py#L106
```pred = F.softmax(pred, dim=1)```
Why  use `softmax` on `pred`?
Is `pred` cls_score here? But in focal_loss and lovasz_loss, there is no `pred = F.softmax(pred, dim=1)`
Could you tell the reason?
Thank you!

**Comment by zhongqiu1245** [comment]
And why set `valid_mask`?
When should I use `valid_mask`?

**Comment by AceFromSpace** [comment]
I am also wondering about that. I am working on image segmentator with 2 classes and after I commented this line Dice Loss staterted to work. Previously I was getting 0 for Acc and IoU for 2nd class

**Comment by PeterVennerstrom** [comment]
valid_mask is the area which should be supervised by the loss function.  The area to exclude from loss calculations (often background) is usually defined as 0 or 255 in a dataset's annotations.   

Most loss functions operate on input between [0, 1].  Softmax and Sigmoid are two common activation functions to convert raw network output (logits) to a [0, 1] interval.  Some Pytorch loss functions like binary_cross_entropy_with_logits or cross_entropy contain the relevant activation function and take logits as input.

**Comment by zhongqiu1245** [comment]
@PeterVennerstrom Thank you!
